pep/app/pods/components/document/text/component.ts

```diff
@@ -467,5 +489,52 @@ export default class DocumentText extends Component<DocumentTextArgs> {
                 });
             }
         });
+
+        if (this.currentUser.preferences?.translationConcordanceEnabled) {
+            const translations = this.containerElement?.querySelectorAll(DocumentTooltipSelectors.TRANSLATION);
+            const loadingTranslation = this.intl.t('common.loading');
+            translations?.forEach((item) => {
+                const paraLangId = item.attributes.getNamedItem('data-lgrid')?.nodeValue;
+                const paraLangRx = item.attributes.getNamedItem('data-lgrx')?.nodeValue;
+
+                const paragraph = item.parentElement;
+                if ((paraLangId || paraLangRx) && paragraph) {
+                    tippy(item, {
+                        appendTo: paragraph,
+                        content: loadingTranslation,
+                        ...this.tippyOptions,
+                        onCreate(instance: DocumentTippyInstance) {
+                            // Setup our own custom state properties - from DOCS
+                            instance._isFetching = false;
+                            instance._loaded = false;
+                        },
+                        onShow: (instance: DocumentTippyInstance) => {
+                            if (instance._isFetching || instance._loaded) {
+                                return;
+                            }
+
+                            this.args
+                                .loadTranslation?.(paraLangId, paraLangRx)
+                                .then(async (text: string) => {
```

Comment:
could you make the `onShow` function `async` as well, and flatten out these promise calls completely?

Reply:
No, this is typed from the package and it's not a promise. So to keep your GreenBuilds 🟢 we need to keep as is